pageserver/src/tenant/mgr.rs

```diff
@@ -1369,7 +1371,7 @@ impl TenantManager {
         &self,
         tenant_shard_id: TenantShardId,
         activation_timeout: Duration,
-    ) -> Result<(), DeleteTenantError> {
+    ) -> Result<StatusCode, DeleteTenantError> {
```

Comment:
Yeah, I don't love having this here either.  This should go away when we finish cleaning up and remove the traditional 202-style path -- I'm using status codes here rather than inventing a new error type that would reduce to "404 or 202".